### PR TITLE
[CU-a73n08, Docs] Glossary - Dispersion

### PR DESCRIPTION
**Pull request by CloudChaoszero**
# Description
* Add "Dispersion"-related definitions in PyMC glossary
* [Optional]: Creating template for adding terms into Glossary

[CU-5t5y0p Glossary](https://app.clickup.com/t/5t5y0p)

### CODE REVIEW

**Review by martinacantaro — COMMENTED** [review]
The Underdispersion entry links to the Overdispersion wikipedia page.

I don't think it's necessary to write the definitions inside quotation marks. 

For the dispersion definition, I would try to clarify what 'streched' and 'squeezed' mean, as it is a bit ambiguous

File: docs/source/glossary.md [hunk 1]
```diff
@@ -12,9 +13,20 @@ Second term
...
   Functional programming is a programming style that prefers the use of basic functions with explicit and distinct inputs and outputs.
   This contrasts with functions or methods that depend on variables that are not explicitly passed as an input (such as accessing `self.variable` inside a method) or that alter the inputs or other state variables in-place, instead of returning new distinct variables as outputs.
+
 Dispatching
   Choosing which function or method implementation to use based on the type of the input variables (usually just the first variable). For some examples, see Python's documentation for the [singledispatch](https://docs.python.org/3/library/functools.html#functools.singledispatch) decorator.
+
+[Dispersion](https://en.wikipedia.org/wiki/Statistical_dispersion)
+  In statistics, dispersion (also called variability, scatter, or spread) is the extent to which a distribution is stretched or squeezed
+[Overdispersion](https://en.wikipedia.org/wiki/Overdispersion)
+  "In statistics, overdispersion is the presence of greater variability (statistical dispersion) in a data set than would be expected based on a given statistical model."
```
Comment (OriolAbril) [inline docs/source/glossary.md]:
```suggestion
  "In statistics, overdispersion is the presence of greater {term}`variability <dispersion>` in a data set than would be expected based on a given statistical model."
```

**Review by OriolAbril — COMMENTED** [review]
Many of these definitions (especially also looking at the issue and other PRs) are quite complicated and many talks have been given and book chapters written about them, so I think it is very important to be generous with links and references.

I have added a suggestion to link to "dispersion" from the definition of overdispersion (same should be done for underdispersion). I think many definitions will need to refer to other terms.

It might also be a good idea to add some references or seealso links. We can discuss that during the meeting, but as this is mostly oriented at newcomers, we could recommend a section of 1-2 books to dive deeper into that term/area which would link the resources repo with this one much more

**Comment by martinacantaro** [comment]
Task linked: [CU-5t5y0p Glossary](https://app.clickup.com/t/5t5y0p)

**Comment by CloudChaoszero** [comment]
Sounds great @martinacantaro. I also added Oriol's recommendation

File: docs/source/glossary.md [hunk 2]
```diff
@@ -3,6 +3,7 @@
 A glossary of common terms used throughout the PyMC3 documentation and examples.
 
 :::::{glossary}
+
 [Term with external reference](https://www.youtube.com/watch?v=6dc7JgR8eI0)
```
Comment (martinacantaro) [inline docs/source/glossary.md]: Remove this section

**Review by martinacantaro — CHANGES_REQUESTED** [review]
Looking good! Solved some conflicts, I'm fine with merging it now and creating some issues to resolve later if there's anything we want to improve upon @OriolAbril . I'd suggest moving links out of the definition titles in the future.

(Edited this issue so although this is categorized "request for changes" type of review, I'm not requesting any changes).

File: docs/source/glossary.md [hunk 3]
```diff
@@ -3,20 +3,18 @@
 A glossary of common terms used throughout the PyMC documentation and examples.
 
 :::::{glossary}
+
```
Comment (OriolAbril) [inline docs/source/glossary.md]: There should be no space between the directive start and it's arguments (i.e. the `:sorted:` thing)

File: docs/source/glossary.md [hunk 4]
```diff
@@ -3,20 +3,18 @@
...
-  to an external reference.
-
-Second term
-  Definitions can have as many lines as desired, and should be written in markdown. Definitions
-  can contain any markdown formatting for MyST to parse, this includes basic markdown like **bold**
-  as well as MyST directives and roles like {fa}`fort awesome,style=fab`
-Functional Programming
-  Functional programming is a programming style that prefers the use of basic functions with explicit and distinct inputs and outputs.
-  This contrasts with functions or methods that depend on variables that are not explicitly passed as an input (such as accessing `self.variable` inside a method) or that alter the inputs or other state variables in-place, instead of returning new distinct variables as outputs.
-Dispatching
```
Comment (OriolAbril) [inline docs/source/glossary.md]: Functional programming and dispatching are actual glossary terms, not examples.

**Review by martinacantaro — APPROVED** [review]

**Comment by CloudChaoszero** [comment]
@martinacantaro are there any additional edits for this? I saw you made some great commits, appreciate it!

**Pull request merged**